### ISSUE TITLE
plot2d: option [xy_scale] has no/strange effect

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#582]
Imported from SourceForge on 2024-07-03 02:00:31
Created by **[mirek-slouf](https://sourceforge.net/u/mirek-slouf/)** on 2016-12-10 09:39:40
Original: https://sourceforge.net/p/maxima/bugs/3260

---

This bug was originally submitted as wxMaxima bug (see the text below), but it appears also in xMaxima:

Sample input: plot2d(x+sin(x),[x,-5,5], [xy_scale, 0.1,1])
Output in xMaxima: in my version, it seems that xy_scale option has no effect
Output in wxMaxima: in my version, the output looks strange => see the attached file

-----
Copy of the original bug report submitted to wxMaxima:
https://github.com/andrejv/wxmaxima/issues/820

Sample input: wxplot2d(x+sin(x),[x,-5,5], [xy_scale, 0.5,1])$

Output is strange in two aspects:

* the scale of the graph is NOT 0.5*x and 1*y
* the size of the cell containing the graph is too large (as if [xy_scale, 1,1] was used)

See the attached files (WXMX with TXT extension and WXMX printed to PDF):

*     WXMX.TXT: test_wxplot2d_xy_scale.wxmx.txt
*     WXMX.PDF: test_wxplot2d_xy_scale.wxmx.pdf

-----
Copy of the answer from gunterkoenigsmann:

That indeed looks strange, even if I am no plot expert and might be wrong. I also can acknowledge that I am able to reproduce this bug. What I can tell is that it also happens with:

plot2d(x+sin(x),[x,-5,5], [xy_scale, 0.5,1])$ 

In this command wxMaxima isn't involved =>I would beg you to report this bug to maxima (https://maxima.sourceforge.net) and will close it for wxMaxima, not because I think it is invalid, but because it is better handled in maxima.

-----
Version of my programs:
wxMaxima version: 16.4.2Maxima version: 5.38.1
Maxima build date: 2016-06-12 17:13:34
Host type: i686-pc-mingw32
System type: Win32 6.1.7601 (Service Pack 1) X86-64
Lisp implementation type: SBCL
Lisp implementation version: 1.3.4

Windows 7 64-bit, PDFCreator 2.4, wxMaxima nightly build 2016-11-30

Attachments:
- https://sourceforge.net/p/maxima/bugs/3260/attachment/test_wxplot2d_xy_scale.wxmx.pdf
- https://sourceforge.net/p/maxima/bugs/3260/attachment/test_wxplot2d_xy_scale.wxmx.txt

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-03 02:00:33
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2016-12-11 01:57:18
Original: https://sourceforge.net/p/maxima/bugs/3260/#e0e7

---

In trying the following example:

~~~~
plot2d ([discrete, [-1, 0.4],[-0.5, 0.2], [0.2, 0.6], [0.8, 0.95]], [xy_scale, FOO, 1.0]);
~~~~

for different values of FOO, it seems that when FOO is smaller than 0.2, Gnuplot draws what seems to be an erroneous picture. Maybe this is a bug in Gnuplot? 

I think it is significant that `set size ratio 0.75` is in Maxima's Gnuplot output file; without that, Gnuplot draws a tall, skinny picture when `set size 0.1, 1.0` is given. Is Gnuplot handling the interaction between `set size ratio 0.75` and `set size 0.1, 1.0` incorrectly?

After trying the example given above:

~~~~
plot2d(x+sin(x),[x,-5,5], [xy_scale, 0.5,1], [plot_format, gnuplot]);
~~~~

I edited the Gnuplot output file to remove the line `set size ratio 0.75`. Then the plot that is displayed does seem to be twice as tall as it is wide, as expected.

Maybe Maxima should omit `set size ratio 0.75` if xy_ratio is present?

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-03 02:00:36
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2016-12-14 05:08:50
Original: https://sourceforge.net/p/maxima/bugs/3260/#b2aa

---

Here is a proposed patch which suppresses `set size ratio 0.75` when `xy_scale` is present:

~~~~
$ git diff src/gnuplot_def.lisp
diff --git a/src/gnuplot_def.lisp b/src/gnuplot_def.lisp
index 458d47a..37b20ac 100644
--- a/src/gnuplot_def.lisp
+++ b/src/gnuplot_def.lisp
@@ -339,7 +339,10 @@
           (format dest "set size ratio -1~%")
           (if (getf plot-options :yx_ratio)
               (format dest "set size ratio ~,8f~%" (getf plot-options :yx_ratio))
-              (format dest "set size ratio 0.75~%")))
+              (if (not (getf plot-options :xy_scale))
+                ;; set size ratio seems to interfere with xy_scale,
+                ;; so emit the default only if there is no xy_scale specified.
+                (format dest "set size ratio 0.75~%"))))
       (if (and (getf plot-options :xy_scale)
                (listp (getf plot-options :xy_scale)))
           (format dest "set size ~{~,8f~^, ~}~%" (getf plot-options :xy_scale))))
~~~~

I've attached 4 figures (bug3260-with-patch-xy_scale-xx-yy.pdf with xx = 1 or 05 and yy = 1 or 05) in which I attempt to replicate the examples given in test_wxplot2d_xy_scale.wxmx.pdf. What does anyone think about those pictures? Do they look acceptable now?



Attachments:
- https://sourceforge.net/p/maxima/bugs/_discuss/thread/6da0f8c4/b2aa/attachment/bug3260-with-patch-xy_scale-05-05.pdf
- https://sourceforge.net/p/maxima/bugs/_discuss/thread/6da0f8c4/b2aa/attachment/bug3260-with-patch-xy_scale-05-1.pdf
- https://sourceforge.net/p/maxima/bugs/_discuss/thread/6da0f8c4/b2aa/attachment/bug3260-with-patch-xy_scale-1-05.pdf
- https://sourceforge.net/p/maxima/bugs/_discuss/thread/6da0f8c4/b2aa/attachment/bug3260-with-patch-xy_scale-1-1.pdf

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-03 02:00:40
Created by **[mirek-slouf](https://sourceforge.net/u/mirek-slouf/)** on 2016-12-18 21:29:53
Original: https://sourceforge.net/p/maxima/bugs/3260/#3290

---

The plots look quite Ok now. Thank you for the patch.

However, I have one additional question: is it possible to make the whole plot smaller (let us say half of the size of the original plot)?

Now the plots themselves are smaller and with correct size ratio, but the total size of the ploting window/size remains the same.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-03 02:00:43
Created by **[villate](https://sourceforge.net/u/villate/)** on 2017-11-20 13:20:58
Original: https://sourceforge.net/p/maxima/bugs/3260/#360d

---

- **labels**:  --> plotting
- **status**: open --> closed

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-03 02:00:47
Created by **[villate](https://sourceforge.net/u/villate/)** on 2017-11-20 13:20:59
Original: https://sourceforge.net/p/maxima/bugs/3260/#b715

---

It would be very hard to change the size of the plotting window, since the way it must be done depends on the gnuplot terminal being used, and plot2d/plot3d attempt to be terminal independent and make use of whatever the default it's in each system.
The examples I've seen in this thread would work better by using option yx_ratio instead of  xy_scale as I have just added in the manual. I have also committed the patch shown by Robert (commit [534f80]) and I'm closing this report.
